client/src/containers/SettingsTab/index.jsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import { Grid, Menu, Header, Divider, Input, Button, Message, Loader } from 'semantic-ui-react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { fetchRepoSettings } from '../../routines/routines';
+import { renameRepo, deleteRepo } from './actions';
+import styles from './styles.module.scss';
+
+const renderDangerField = (header, description, buttonName, clickHandler) => (
+  <Message className={styles.dangerField}>
+    <Grid>
+      <Grid.Column width={12}>
+        <Header as="h6">{header}</Header>
+        <p>{description}</p>
+      </Grid.Column>
+      <Grid.Column width={4}>
+        <Button type="button" className={styles.dangerButton} fluid onClick={clickHandler}>
+          {buttonName}
+        </Button>
+      </Grid.Column>
+    </Grid>
+  </Message>
+);
+
+class RepoSettings extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const {
+      match: {
+        params: { owner, repository }
+      }
+    } = this.props;
+    this.oldName = repository;
+
+    this.state = {
+      name: repository,
+      owner
+    };
+  }
+
+  componentDidMount() {
+    const { owner, name } = this.state;
+    this.props.fetchRepoSettings({
+      owner,
+      name
+    });
+  }
+
+  onClickDelete = () => {
+    const { owner, name } = this.state;
+    this.props.deleteRepo({
+      owner,
+      name
+    });
+  };
+
+  handleChangeRepoName = event => {
+    this.setState({
+      name: event.target.value
+    });
+  };
+
+  onClickRename = () => {
+    const { name, owner } = this.state;
+    const { oldName } = this;
+
+    this.props.renameRepo({
```

Comment:
destructurizing

Reply:
In this manner?
```suggestion
    const { oldName, props } = this;
    props.renameRepo({
```
or:
```suggestion
    const { oldName, props: {renameRepo} } = this;
    renameRepo({
```